tests/cpp/test_gpu1.cpp

```diff
@@ -5388,180 +5388,169 @@ TEST_F(NVFuserTest, FusionReductionSchedulerMultiDimFastest_CUDA) {
+  TensorView* tv1 = sum(tv0_cast, {0});
 
-  // Tried to cut down the number iterations with just
-  // doing every other power of 2.
-  for (int i = 1; i <= 1024 * 1024; i <<= 2) {
-    red_dims.push_back(i);
+  TensorView* tv1_cast = tv1;
+  if (is_fp16) {
+    tv1_cast = castOp(DataType::Half, tv1);
+  }
+  if (is_bf16) {
+    tv1_cast = castOp(DataType::BFloat16, tv1);
   }
 
-  for (auto dtype : dtypes) {
-    at::ScalarType aten_dtype = data_type_to_aten(dtype);
-    for (auto& rdim : red_dims) {
-      // Shmoo tests can occupy a lot of memory due to allocating many
-      // different tensor sizes. So in order to avoid an OOM during this
-      // test, we manually clear the allocator after it's reached a certain
-      // threshold.
-      maybeClearAllocator();
+  fusion.addOutput(tv1_cast);
 
-      Fusion fusion;
-      FusionGuard fg(&fusion);
+  auto options = at::TensorOptions().dtype(aten_dtype).device(at::kCUDA, 0);
 
-      bool is_fp16 = dtype == DataType::Half;
-      bool is_bf16 = dtype == DataType::BFloat16;
+  at::Tensor aten_input = at::randn({rdim}, options);
+  auto aten_output = aten_input.to(at::kDouble).sum({0});
 
-      TensorView* tv0 = makeSymbolicTensor(1, dtype);
-      fusion.addInput(tv0);
+  auto reduction_params = getReductionHeuristics(&fusion, {aten_input});
+  NVF_CHECK(reduction_params != nullptr, "Reduction is not found!");
+  scheduleReduction(&fusion, *reduction_params);
+  auto lparams = reduction_params->lparams;
 
-      TensorView* tv0_cast = tv0;
-      if (is_fp16 || is_bf16) {
-        tv0_cast = castOp(DataType::Float, tv0);
-      }
+  FusionExecutor fe;
+  fe.compileFusion(&fusion, {aten_input}, lparams);
+  auto cg_outputs = fe.runFusion({aten_input}, lparams);
 
-      TensorView* tv1 = sum(tv0_cast, {0});
+  testValidate(
+      &fusion,
+      cg_outputs,
+      {aten_input},
+      {aten_output},
+      __LINE__,
+      __FILE__,
+      "",
+      lparams);
+}
+INSTANTIATE_TEST_SUITE_P(
+    ,
+    ReductionNoODimShmoo,
+    ::testing::Combine(
+        testing::ValuesIn(getFloatingDataTypes()), // data type
+        testing::ValuesIn(Pow2Vals1to1Million)), // reduction dimension size
+    [](const testing::TestParamInfo<ReductionNoODimParams>& info)
+        -> std::string {
+      std::stringstream ss;
+      ss << "dtype_" << std::get<0>(info.param);
+      ss << "_rsize_" << std::get<1>(info.param);
+      return sanitizeTestName(ss.str());
+    });
 
-      TensorView* tv1_cast = tv1;
-      if (is_fp16) {
-        tv1_cast = castOp(DataType::Half, tv1);
-      }
-      if (is_bf16) {
-        tv1_cast = castOp(DataType::BFloat16, tv1);
-      }
+// data type, reduction dim size, iteration dim size, reduction axis
+using ReductionWithIterDimParams =
+    std::tuple<DataType, int64_t, int64_t, int64_t>;
+using ReductionWithIterDimShmoo =
+    NVFuserFixtureParamTest<ReductionWithIterDimParams>;
+TEST_P(ReductionWithIterDimShmoo, FusionReductionSchedulerDimShmoo_CUDA) {
```

Comment:
We could probably remove `Shmoo` (and `_CUDA`) from the name. Something like this?
```suggestion
TEST_P(ReductionWithIterDim, Test) {
```